lib/org/openpsa/sales/sales/pdf.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * @package org.openpsa.sales
+ * @copyright CONTENT CONTROL http://www.contentcontrol-berlin.de/
+ * @license http://www.gnu.org/licenses/gpl.html GNU General Public License
+ */
+
+/**
+ * PDF Manager
+ *
+ * @package org.openpsa.sales
+ */
+class org_openpsa_sales_sales_pdf
+{
+    /**
+     * @var org_openpsa_sales_salesproject_dba
+     */
+    private $salesproject;
+
+    public function __construct(org_openpsa_sales_salesproject_dba $salesproject)
+    {
+        $this->salesproject = $salesproject;
+    }
+
+    /**
+     * @return midcom_db_attachment|NULL
+     */
+    public function get_attachment()
+    {
+        $pdf_files = org_openpsa_helpers::get_dm2_attachments($this->salesproject,"pdf_file");
+        if (!empty($pdf_files)) {
+            return reset($pdf_files);
+        }
+        return $this->render_and_attach();
```

Comment:
this is a behavior change compared to the previous version: Now you always render the PDF if it is not found. This is fine by me, but if you want to keep it that way, the phpdoc needs to be adapted and also the calling code further down in this class